tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rules/FunAppRule.scala

```diff
@@ -85,16 +90,20 @@ class FunAppRule(rewriter: SymbStateRewriter) extends RewritingRule {
     }
   }
 
+  private def getFieldName(argEx: TlaEx): String = {
+    argEx match {
+      case ValEx(TlaStr(key)) => key
+      case _ => throw new RewriterException(s"Accessing a record with a non-constant key $argEx", argEx)
```

Comment:
iiu, it's not just this is non-constant, but that it's not a possible valid field? So maybe:

```suggestion
      case _ => throw new RewriterException(s"Accessing a record with value that cannot be a key $argEx", argEx)
```

?